### PR TITLE
Update to support spark 1.4

### PR DESCRIPTION
**Pull request by adrian-wang**

### CODE REVIEW

**Comment by GraceH** [comment]
Thanks @adrian-wang . The shims way looks good to support multiple version. Meanwhile, I have one concern about the changeable API in Spark sql/streaming. 

  1. If we need to support a new version, I am ok to provide a new shim. But do we need to change the guard API in other files?
  2. The old shim file itself will be updated accordingly?

**Comment by GraceH** [comment]
We'd better to have some API designs as the indicator to show 
1. how to add new shim support for streaming sql
2. if a new version doesn't introduce any change, what shall we do?

**Comment by GraceH** [comment]
Add @chenghao-intel for more comments.

**Comment by chenghao-intel** [comment]
As we discussed offline, in order to keep the less duplicated code as much as possible, instead of the put entire class into the Shim code, a better idea is to move the in-compatible methods into Shim trait, and we can inherits from the trait as needed.

File: project/Build.scala [hunk 1]
```diff
@@ -23,7 +23,8 @@ import org.scalastyle.sbt.ScalastylePlugin
 import scalariform.formatter.preferences._
 
 object Properties {
-  val SPARK_VERSION = "1.3.0"
+  val SPARK_VERSION = "1.4.0"
+  val SPARK_SHIM_VERSION = SPARK_VERSION
```
Comment (chenghao-intel) [inline project/Build.scala]: Can we remove this line? Seems only used once in the `Build.scala`

**Comment by chenghao-intel** [comment]
I am also have concern on supporting different version of Spark, as a POC, do we really need to support both versions (1.3 & 1.4)? And spark 1.5 will be release soon, that's probably give us some burden to maintain the code.

File: project/Build.scala (hunk 1 above)
Comment (adrian-wang) [inline project/Build.scala]: This is for version update. If APIs we used stays the same between 1.6 and 1.5, we could assign SPARK_VERSION="1.6.0" and use SPARK_SHIM_VERSION="1.5.0"

**Comment by adrian-wang** [comment]
@chenghao-intel @GraceH I've updated this.

**Comment by GraceH** [comment]
1. It seems it is still possible to change 1.3 and 1.4 while we have new versions to support. 
2. How can we make it easy to follow to create new shims?

**Comment by happycherry** [comment]
Build finished. No test results found.

**Pull request closed without merging**